### ISSUE TITLE
[ Release: File Not Found ] Download link on release page does not work

### ISSUE THREAD

**Issue by sa-tasche** [JulNadeauCA/libagar#86]
When I click on the link to the latest release I see the error 404.
Both zip files are affected.

**Comment by JulNadeauCA** [comment]
Thanks for pointing this out. We have been having some trouble with GitHub releases recently. I'll see what I can do to fix it. For now, you can download all official releases from the website at:
[https://libagar.org/download.html](https://libagar.org/download.html)

**Comment by GitMensch** [comment]
Seems all releases are gone from GitHub - and there are also no git tags here either.
Can those please restored?

That will also help in case of the website not being reachable via curl #60.

**Comment by JulNadeauCA** [comment]
The problem is that as of about a year ago, the normal method for pushing our tags to the GitHub repository stopped working and I could not find the reason why.

Our master repository is the SVN repository over at https://dev.csoft.net/. You can see the tags at: https://dev.csoft.net/agar/tags/.

I have a working copy checked out under `~/git/agar-gitsvn` for purposes of running `git push`. It used to be that `git push origin --tags` would do the trick. However, it no longer appears to do anything:
```
vedge@resin(~/git/agar-gitsvn)$ git push origin --tags
Username for 'https://github.com': 
Password for 'https://JulNadeauCA@github.com': 
Everything up-to-date
```

If I try to push a specific tag, I get the error:
```
vedge@resin(~/git/agar-gitsvn)$ git push origin release-1.7.1
error: src refspec release-1.7.1 does not match any
error: failed to push some refs to 'https://github.com/JulNadeauCA/libagar.git'
```

I would love to get this fixed, but I don't know what I'm doing wrong.

**Comment by GitMensch** [comment]
note sure, but does `git svn fetch --tags=tags` work?
If not you could recreate them via script (I'd ask one of the LLMs to create the script) or redo the git2svn (but then all forks and PRs would post-break because of the different hash).

Just out of interest from one maintainer of upstream in svn to another: do you also plan on moving to git?

We'll do a git2svn (and likely recreate the svn to combine our "history" with our "code" repo directly before), then push to codeberg.org (+ mirror to GitHub for CI only running here, like IBM sponsored ones + let others "more easily" contribute that are only here).

**Comment by JulNadeauCA** [comment]
That doesn't seem to work:
```
vedge@resin(~/git/agar-gitsvn)$ git svn fetch --tags=tags
Unknown option: tags
```
I've also tried `git push --follow-tags` without success. `git branch` does list the branches correctly:
```
vedge@resin(~/git/agar-gitsvn)$ git branch -r
  agar14
  origin/trunk
  tags/release-1.0
  tags/release-1.1
  tags/release-1.2
  tags/release-1.3
  tags/release-1.3.2
  tags/release-1.3.3
  tags/release-1.3.4
  tags/release-1.4.0
  tags/release-1.4.1
  tags/release-1.5.1
  tags/release-1.6.0
  tags/release-1.7.0
  tags/release-1.7.1
  trunk
```

**Comment by JulNadeauCA** [comment]
> Just out of interest from one maintainer of upstream in svn to another: do you also plan on moving to git?
> 
> We'll do a git2svn (and likely recreate the svn to combine our "history" with our "code" repo directly before), then push to codeberg.org (+ mirror to GitHub for CI only running here, like IBM sponsored ones + let others "more easily" contribute that are only here).

We have no plans to move to git in the short term since svn has been reliable and provides everything we need, but we might switch to git in the future if we run into the need for specific features not in svn.

**Comment by JulNadeauCA** [comment]
If I run `git tag -l`, it returns an empty list. I'm wondering if the `tags` mapping thing could be the issue. My git config is as follows:
```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[svn-remote "svn"]
        url = file:///home/dev/agar
        fetch = trunk:refs/remotes/trunk
        branches = branches/*:refs/remotes/*
        tags = tags/*:refs/remotes/tags/*
[svn]
        authorsfile = /home/vedge/git/authors.txt
[remote "origin"]
        url = https://github.com/JulNadeauCA/libagar.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "trunk"]
        remote = origin
        merge = refs/heads/trunk
```

**Comment by JulNadeauCA** [comment]
Aha! The following command fixed my problem:
```git for-each-ref --format='%(refname:short)' refs/remotes/tags | cut -d/ -f2 | while read tag; do git tag "$tag" "remotes/tags/$tag"; done```

The tags should be appearing again on GitHub:
```
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 266 bytes | 266.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
To https://github.com/JulNadeauCA/libagar.git
 * [new tag]             release-1.0 -> release-1.0
 * [new tag]             release-1.1 -> release-1.1
 * [new tag]             release-1.2 -> release-1.2
 * [new tag]             release-1.3 -> release-1.3
 * [new tag]             release-1.3.2 -> release-1.3.2
 * [new tag]             release-1.3.3 -> release-1.3.3
 * [new tag]             release-1.3.4 -> release-1.3.4
 * [new tag]             release-1.4.0 -> release-1.4.0
 * [new tag]             release-1.4.1 -> release-1.4.1
 * [new tag]             release-1.5.1 -> release-1.5.1
 * [new tag]             release-1.6.0 -> release-1.6.0
 * [new tag]             release-1.7.0 -> release-1.7.0
 * [new tag]             release-1.7.1 -> release-1.7.1
```

**Comment by GitMensch** [comment]
yes - this issue can be closed now - looking forward to 1.7.2 ... nope, the version is already on 1.8.0 - so looking forward to that :-)
